module/user-extension/helper.js

```diff
@@ -538,169 +573,307 @@ module.exports = class UserExtensionHelper {
+                                { $skip: pageSize * (pageNo - 1) },
+                                { $limit: pageSize }
+                            ],
+                        }
+                    }, {
+                        $project: {
+                            "data": 1,
+                            "count": {
+                                $arrayElemAt: ["$totalCount.count", 0]
+                            }
+                        }
                     }
+                ];
+
+                let userExtensionData =
+                    await database.models.userExtension.aggregate(queryObject);
+
+            
+                let entityDocuments = [];
+                if (userExtensionData[0]) {
+                    userExtensionData[0].data.forEach(entity => {
+                        entityDocuments.push({
+                            _id: entity.entityDocuments._id,
+                            name: entity.entityDocuments.metaInformation.name,
+                        });
+                    });
+                    return resolve({ data:entityDocuments,count:userExtensionData[0].count })
+
+                } else {
+                    return resolve({
+                        status: httpStatusCode['bad_request'].status,
+                        message: constants.apiResponses.USER_ENTITIES_NOT_FOUND
+                    });
                 }
+            } catch (error) {
+                return reject(error);
             }
+        })
+    }
 
-            if ( userExtensionData[0].roleDocuments && userExtensionData[0].roleDocuments.length > 0 ) {
-
-                userExtensionData[0].roleDocuments.forEach(role => {
-                    roleMap[role._id.toString()] = role;
-                })
-                let entityMap = {};
-                
-                userExtensionData[0].entityDocuments.forEach(entity => {
-                    entity.metaInformation.childrenCount = 0;
-                    entity.metaInformation.entityType = entity.entityType;
-                    entity.metaInformation.entityTypeId = entity.entityTypeId;
-                    entity.metaInformation.subEntityGroups = new Array;
-
-                    Array.isArray(enityTypeToImmediateChildrenEntityMap[entity.entityType]) && enityTypeToImmediateChildrenEntityMap[entity.entityType].forEach(immediateChildrenEntityType => {
-                        if (entity.groups && entity.groups[immediateChildrenEntityType]) {
-                            entity.metaInformation.immediateSubEntityType = immediateChildrenEntityType;
-                            entity.metaInformation.childrenCount = entity.groups[immediateChildrenEntityType].length;
-                        }
-                    })
+    /**
+       * Get user entity types
+       * @method
+       * @name getEntityTypes
+       * @param {Object} filterQueryObject - filtered data.
+       * @returns {Object} 
+       */
+
+    static getEntityTypes(filterQueryObject) {
```

Comment:
@aman-tunerlabs - Please expect userId here, status etc filters should be setup here